### ISSUE TITLE
How to generate an anomaly map？

### ISSUE THREAD

**Issue by aldazhu** [DonaldRR/SimpleNet#1]
I have trained the model following the instructions in main.py, but how can I generate an anomaly map? I couldn't find this part in the project,  I'm not sure if I missed it.

**Comment by Liuwenyu123** [comment]
缺少测试文件test.py

**Comment by bbmanboy** [comment]
我也没找到相关可视化的代码，有一个阈值optimal_threshold，segmentations可以自己生成一下结果图

**Comment by DonaldRR** [comment]
We didn't include the full visualization code, but you could still visualize the anomaly map adding few lines of code in the evaluation process.

**Comment by thanhthanhhp123** [comment]
> We didn't include the full visualization code, but you could still visualize the anomaly map adding few lines of code in the evaluation process.

can you help we to do this? please

**Comment by William9Baker** [comment]
> > We didn't include the full visualization code, but you could still visualize the anomaly map adding few lines of code in the evaluation process.
> 
> can you help we to do this? please

Did you solve it? I add the code of saving image, but it didn't work. The images I saved are not 0-1 binary images.

**Comment by liliujin** [comment]
> > > We didn't include the full visualization code, but you could still visualize the anomaly map adding few lines of code in the evaluation process.
> > 
> > 
> > can you help we to do this? please
> 
> Did you solve it? I add the code of saving image, but it didn't work. The images I saved are not 0-1 binary images.

Hi，so is your problem solved?can you help me to do this?I will thank you very much.

**Comment by thanhthanhhp123** [comment]
> > > > We didn't include the full visualization code, but you could still visualize the anomaly map adding few lines of code in the evaluation process.
> > > 
> > > 
> > > can you help we to do this? please
> > 
> > 
> > Did you solve it? I add the code of saving image, but it didn't work. The images I saved are not 0-1 binary images.
> 
> Hi，so is your problem solved?can you help me to do this?I will thank you very much.

I have solved it contact me with my email I can help you

**Comment by Abhijeet99999** [comment]
@thanhthanhhp123 , Contacted you over email, Can you help please to visualized the results ?

**Comment by Abhijeet99999** [comment]
@liliujin , Can you help please to visualized the results ?

**Comment by thanhthanhhp123** [comment]
You can see my repository on Github:
https://github.com/thanhthanhhp123/SimpleNetv2
After training you can change "test" variable this to generate anomaly map:




*if not test:                i_auroc, p_auroc, pro_auroc =
SimpleNet.train(dataloaders["training"], dataloaders["testing"])
else:                i_auroc, full_pixel_auroc =
 SimpleNet.test(dataloaders["training"], dataloaders["testing"],
save_segmentation_images)*

**Comment by Abhijeet99999** [comment]
Hello Tran,

Greetings :)

Thank you, for your help. Do you have inference test file which can test testing sample ?

Best,
Abhijeet Thakare


-----Original Message-----

**Comment by thanhthanhhp123** [comment]
In the dataset file, we split the dataset to training and test set. After
training, you only change the test variable to gen anomaly map. After that
you can have results like this.
[image: image.png]

**Comment by Abhijeet241093** [comment]
Whatever, you're saying followed. Wrote inference pipline but didn't see any visulization results. Nor found https://github.com/thanhthanhhp123/SimpleNetv2https://github.com/thanhthanhhp123/SimpleNetv2 this is helpful.

**Comment by Abhijeet241093** [comment]
Trained model without test data and ground truth and after that designed inference_pipline to predict inference for test sample without using ground truth, howhevere, its wastage of time, number of errors of  simplenet.py coming, which took much time and become unresolved.
